### PR TITLE
8307305: Update debugger tests to support JTREG_TEST_THREAD_FACTORY mode

### PR DESCRIPTION
**Pull request by lmesnik**
The debugger tests might start all debugee using virtual threads when property "main.wrapper" is set.
However, the new mode JTREG_TEST_THREAD_FACTORY do the same thing. The only difference is that it use predefined problemlist names and doesn't set property "main.wrapper" as a part of jtreg properties. So nsk wrapper should set it additionally.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307305](https://bugs.openjdk.org/browse/JDK-8307305): Update debugger tests to support JTREG_TEST_THREAD_FACTORY mode

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13763/head:pull/13763` \
`$ git checkout pull/13763`

Update a local copy of the PR: \
`$ git checkout pull/13763` \
`$ git pull https://git.openjdk.org/jdk.git pull/13763/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13763`

View PR using the GUI difftool: \
`$ git pr show -t 13763`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13763.diff">https://git.openjdk.org/jdk/pull/13763.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13763#issuecomment-1532177308)

### CODE REVIEW

**Comment by plummercj** [comment]
What about com/sun/jdi tests?

File: test/hotspot/jtreg/vmTestbase/nsk/share/MainWrapper.java [hunk 1]
```diff
@@ -46,7 +46,8 @@ public static void main(String[] args) throws Throwable {
         Finalizer finalizer = new Finalizer(new FinalizableObject());
         finalizer.activate();
 
-
+        // Some tests use this property to understand if virtual threads are used
+        System.setProperty("main.wrapper", "Virtual");
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/share/MainWrapper.java:50]: Shouldn't this be:

`System.setProperty("main.wrapper", wrapperName);`

**Comment by plummercj** [comment]
I think MainWrapper.java could use some documentation on why/how/when it used. This is something that should have been done before first committing it when loom changes were integrated.

Also, I see what looks like a bug w.r.t. this code in Launcher.java:

                if (System.getProperty("main.wrapper") != null) {
                    cmdline = MainWrapper.class.getName() + " " + System.getProperty("main.wrapper") +  " " + cmdline;
                }

It gets the main.wrapper property in order to pass it to MainWrapper.main(), which is in charge of setting the properly, so how could it ever already be set when this Launcher.java code is executed? Same thing in Binder.java and DebugeeBinder.java.

**Comment by lmesnik** [comment]
> I think MainWrapper.java could use some documentation on why/how/when it used. This is something that should have been done before first committing it when loom changes were integrated.
> 
> Also, I see what looks like a bug w.r.t. this code in Launcher.java:
> 
> ```
>             if (System.getProperty("main.wrapper") != null) {
>                 cmdline = MainWrapper.class.getName() + " " + System.getProperty("main.wrapper") +  " " + cmdline;
>             }
> ```
> 
> It gets the main.wrapper property in order to pass it to MainWrapper.main(), which is in charge of setting the properly, so how could it ever already be set when this Launcher.java code is executed? Same thing in Binder.java and DebugeeBinder.java.

let me file sub-task to add documentation about using "main.wrapper" as a part of https://bugs.openjdk.org/browse/JDK-8303773.

Currently, it is propagated to debugee as a part of `test.vm.opts` . However the goal is to use `JTREG_TEST_THREAD_FACTORY` which set Virtual test thread factory and set all required properties. So it is needed to manually set them in debugee while launching them using nsk Wrapper or TestScaffold.

File: test/hotspot/jtreg/vmTestbase/nsk/share/MainWrapper.java (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/share/MainWrapper.java]: Thanks, better to use wrapperName as a well as to set it it in TestScaffold.

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**